subscriptions/src/test/subscriptions.spec.ts

```diff
@@ -0,0 +1,191 @@
+import test from 'ava';
+import * as im from 'immutable';
+import { stub } from 'sinon';
+import { concat, EMPTY, NEVER, of, Subject } from 'rxjs';
+import { delay, take, tap, toArray } from 'rxjs/operators';
+import { Patch, Version } from '@binaris/shift-interfaces/dist/subscriptions';
+import {
+  DEREGISTER,
+  NonEmptyArray,
+  poll,
+  Poller,
+  PollerInputWithObserver,
+  pollReducer,
+  publish,
+  publishInputMapper,
+  REGISTER,
+  serverUpdates,
+} from '../subscriptions';
+
+// tslint:disable-next-line:variable-name
+const Version = (major: number, minor: number): Version => [major, minor];
+const eqVersion = (v1: Version, v2: Version): boolean => v1[0] === v2[0] && v1[1] === v2[1];
+const mockPoller = () => stub<Parameters<Poller>, ReturnType<Poller>>().resolves([]);
+
+test('poll polls when asked to register key', async (t) => {
+  const poller = mockPoller();
+  await of(
+    { action: REGISTER, key: 'test', version: Version(1, 1) },
+  ).pipe(
+    poll(poller),
+  ).toPromise();
+  t.is(poller.args.length, 1);
+  t.deepEqual(poller.args[0][0].toJS(), { test: Version(1, 1) });
+});
+
+test('poll stops polling when asked to deregister key', async (t) => {
+  const poller = mockPoller();
+  await of(
+    { action: REGISTER, key: 'test1', version: Version(1, 1) },
+    { action: REGISTER, key: 'test2', version: Version(1, 2) },
+    { action: DEREGISTER, key: 'test1' },
+  ).pipe(
+    poll(poller),
+  ).toPromise();
+  t.is(poller.args.length, 3);
+  t.deepEqual(poller.args[0][0].toJS(), { test1: Version(1, 1) });
+  t.deepEqual(poller.args[1][0].toJS(), { test1: Version(1, 1), test2: Version(1, 2) });
+  t.deepEqual(poller.args[2][0].toJS(), { test2: Version(1, 2) });
+});
+
+test('poll updates version on poll result', async (t) => {
+  const poller = mockPoller().resolves([['test', [{ version: Version(1, 2), ops: [] }]]]);
```

Comment:
Not sure we want to commit to a version update on an idempotent write.  Maybe change the stored value too, to make that clear?